apps/api/auth.R

```diff
@@ -0,0 +1,81 @@
+#* Obtain the encrypted password for a user
+#* @param username Username, which is also the 'salt'
+#* @param password Unencrypted password
+#* @param secretkey Secret Key, which if null, is set to 'notasecret'
+#* @return Encrypted password
+#* @author Tezan Sahu
+get_crypt_pass <- function(username, password, secretkey = NULL) {
+  secretkey <- if(is.null(secretkey)) "notasecret" else secretkey
+  dig <- secretkey
+  salt <- username
+  for (i in 1:10) {
+    dig <- digest::digest(
+      paste(dig, salt, password, secretkey, sep="--"), 
+      algo="sha1", 
+      serialize=FALSE
+    )
+  }
+  return(dig)
+}
+
+
+
+#* Check if the encrypted password for the user is valid
+#* @param username Username
+#* @param crypt_pass Encrypted password
+#* @return TRUE if encrypted password is correct, else FALSE
+#* @author Tezan Sahu
+validate_crypt_pass <- function(username, crypt_pass) {
+  settings <-list(database = list(bety = list(
+    driver = "PostgreSQL", 
+    user = "bety", 
+    dbname = "bety", 
+    password = "bety", 
+    host="postgres"
```

Comment:
are you using docker? couldn't host be different, maybe it is best not to have hardcoded db settings?

Reply:
As of now, to run locally (without docker), you can just modify `host="postgres` to `host=localhost`.

Reply:
@robkooper Do let me know your thoughts about how we could parameterize the host if possible.

Reply:
Changed db connection at all points to use `betyConnect()`